src/arrayMethodReduce.js

```diff
@@ -5,7 +5,16 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let startValue = 0;
+    if (initialValue === undefined) {
+      initialValue = this[0];
```

Comment:
I'd recommend avoiding overriding function arguments. Better to define a local variable along with `startValue` like `result = initialValue`